bundles/org.openhab.binding.solarman/src/main/java/org/openhab/binding/solarman/internal/SolarmanLoggerHandler.java

```diff
@@ -0,0 +1,223 @@
+ *
+ * @author Catalin Sanda - Initial contribution
+ */
+@NonNullByDefault
+public class SolarmanLoggerHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(SolarmanLoggerHandler.class);
+
+    private final DefinitionParser definitionParser;
+    private final SolarmanChannelManager solarmanChannelManager;
+    private final LoggerState loggerState;
+    @Nullable
+    private volatile ScheduledFuture<?> scheduledFuture;
+
+    public SolarmanLoggerHandler(Thing thing) {
+        super(thing);
+        this.definitionParser = new DefinitionParser();
+        this.solarmanChannelManager = new SolarmanChannelManager();
+        this.loggerState = new LoggerState();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        SolarmanLoggerConfiguration config = getConfigAs(SolarmanLoggerConfiguration.class);
+        SolarmanLoggerConnector solarmanLoggerConnector = new SolarmanLoggerConnector(config);
+
+        List<Channel> staticChannels = thing.getChannels().stream()
+                .filter(channel -> !channel.getProperties().containsKey(DYNAMIC_CHANNEL)).toList();
+
+        InverterDefinition inverterDefinition = definitionParser.parseDefinition(config.inverterType);
+
+        if (inverterDefinition == null) {
+            logger.error("Unable to find a definition for the provided inverter type");
+            updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Unable to find a definition for the provided inverter type");
+            return;
+        } else {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Found definition for {}", config.inverterType);
+            }
+        }
+        SolarmanV5Protocol solarmanV5Protocol = new SolarmanV5Protocol(config);
+
+        String localAdditionalRequests = config.getAdditionalRequests();
+        String additionalRequests = localAdditionalRequests != null ? localAdditionalRequests : "";
+
+        List<Request> mergedRequests = StringUtils.isNotEmpty(additionalRequests)
+                ? mergeRequests(inverterDefinition.getRequests(), extractAdditionalRequests(additionalRequests))
+                : inverterDefinition.getRequests();
+
+        Map<ParameterItem, ChannelUID> paramToChannelMapping = mergeMaps(
+                extractChannelMappingFromChannels(staticChannels),
+                setupChannelsForInverterDefinition(inverterDefinition));
+
+        SolarmanChannelUpdater solarmanChannelUpdater = new SolarmanChannelUpdater(this::updateState);
+
+        scheduledFuture = scheduler.scheduleWithFixedDelay(() -> {
+            boolean fetchSuccessful = solarmanChannelUpdater.fetchDataFromLogger(mergedRequests,
+                    solarmanLoggerConnector, solarmanV5Protocol, paramToChannelMapping, loggerState);
```

Comment:
Communication between the `ThingHandler` and the `SolarmanChannelUpdater` calls is very limited when using this boolean. A common approach is to use assume a requests is succesfull and throw a custom exception otherwise. 

For example a `SolarmanCommunicationException` or a `SolarmanAuthenticationException`
This would make it possible to differentiate between transient network errors and configuration or authentication errors. 
That information can be used to set the ThingState with details, providing the user with usefull information.

Reply:
Updated the code to use an object which conveys more information which is given to the user. Internally it uses a bunch of custom exceptions.